File: lectures/8.documentation_readability/example_code.R
"""
This is a docstring. It describes the purpose of the file.
In our case, this file will demonstate how to document code.

Note: Any formatting errors in this file will be highlighted
by the linter and fixed by ruff (the auto-formatter).
"""

library(dplyr)

# An example of a code comment
# Category A analyzed/visualized separately
data_df <- data_df %>%  
    dplyr::filter(column_one != “category_a”)  

# An example of an in-line comment
long_df <- transform_tidy_long(data_df)  # prepare viz

File: materials/08.documentation/example_code.py
"""
This is a docstring. It describes the purpose of the file.
In our case, this file will demonstate how to document code.

Note: Any formatting errors will be fixed by pre-commit.

To test this file, run `pytest .` in terminal :)
"""

import numpy as np
import pandas as pd


def add_integers(a: int, b: int) -> int:
    """Add two integers and return the result.

    Args:
      a (int): The first integer.
      b (int): The second integer.

    Returns: int: The sum of a and b.
    """
    return a + b
